src/python/WMCore/WMSpec/Steps/Executors/StageOut.py

```diff
@@ -136,48 +135,40 @@ def execute(self, emulator = None):
             # for that step; or so I hope
             files = stepReport.getAllFileRefsFromStep(step = step)
             for fileName in files:
-                if not hasattr(fileName, 'lfn') and hasattr(fileName, 'pfn'):
-                    # Then we're truly hosed on this file; ignore it
-                    msg = "Not a file: %s" % fileName
+
+                # make sure the file information is consistent
+                if hasattr(fileName, 'pfn') and ( not hasattr(fileName, 'lfn') or not hasattr(fileName, 'module_label') ):
```

Comment:
Hmm, maybe the pfn has already applied the storage.xml rules and that's why we need it?
BTW, there are two unneeded spaces on that line.